.eslintrc

```diff
@@ -0,0 +1,135 @@
+        "@typescript-eslint/no-empty-interface": "off",
+        "@typescript-eslint/no-explicit-any": "off",
+        "@typescript-eslint/no-inferrable-types": "error",
+        "@typescript-eslint/no-misused-new": "error",
+        "@typescript-eslint/no-namespace": "off",
+        "@typescript-eslint/no-object-literal-type-assertion": "off",
+        "@typescript-eslint/no-parameter-properties": "off",
+        "@typescript-eslint/no-this-alias": "error",
+        "@typescript-eslint/no-triple-slash-reference": "off",
+        "@typescript-eslint/no-unnecessary-qualifier": "error",
+        "@typescript-eslint/no-unnecessary-type-assertion": "error",
+        "@typescript-eslint/no-use-before-define": "off",
+        "@typescript-eslint/no-var-requires": "off",
+        "@typescript-eslint/prefer-for-of": "error",
+        "@typescript-eslint/prefer-function-type": "error",
+        "@typescript-eslint/prefer-interface": "error",
+        "@typescript-eslint/prefer-namespace-keyword": "error",
+        "semi": "off",
+        "@typescript-eslint/semi": "error",
+        "@typescript-eslint/type-annotation-spacing": "error",
+        "@typescript-eslint/unified-signatures": "error",
+
+        "microsoft-typescript/object-literal-surrounding-space": "error",
+        "microsoft-typescript/no-type-assertion-whitespace": "error",
+        "microsoft-typescript/type-operator-spacing": "error",
+        "microsoft-typescript/only-arrow-functions": ["error", {
+            "allowNamedFunctions": true ,
+            "allowDeclarations": true
+        }],
+        "microsoft-typescript/no-double-space": "error",
+        "microsoft-typescript/boolean-trivia": "error",
+        "microsoft-typescript/no-in-operator": "error",
+        "microsoft-typescript/debug-assert": "error",
+        "microsoft-typescript/no-keywords": "error",
+
+        "import/no-extraneous-dependencies": "error",
+
+        "no-null/no-null": "error",
+
+        "arrow-body-style": "off",
+        "arrow-parens": "off",
+        "brace-style": ["error", "stroustrup", { "allowSingleLine": true }],
+        "comma-dangle": "off",
+        "complexity": "off",
+        "constructor-super": "error",
+        "curly": ["error", "multi-line"],
+        "dot-notation": "error",
+        "eol-last": "off",
+        "eqeqeq": "error",
+        "guard-for-in": "off",
+        "linebreak-style": ["error", "windows"],
+        "max-classes-per-file": "off",
+        "max-len": "off",
+        "new-parens": "error",
+        "no-bitwise": "off",
+        "no-caller": "error",
+        "no-cond-assign": "off",
+        "no-console": "off",
+        "no-debugger": "off",
+        "no-duplicate-case": "error",
+        "no-duplicate-imports": "error",
+        "no-empty": "error",
+        "no-eval": "error",
+        "no-extra-bind": "error",
+        "no-fallthrough": ["error", { "commentPattern": "falls[\\s\\w]*through" }],
```

Comment:
If I recall correctly, this rule doesn't like case statements separated by comments.
How do we handle this without having to put `// falls through` every time we put a comment between two case statements?

Reply:
I think there is no option to use comments without appropriate **keyword** _(can be added in `commentPattern` option)_. Need to explicitly add the comment // falls through in order to avoid an eslint error.

```typescript
case SyntaxKind.MethodDeclaration:
    if (isObjectLiteralOrClassExpressionMethod(node)) {
        return ContainerFlags.IsContainer | ContainerFlags.IsControlFlowContainer | ContainerFlags.HasLocals | ContainerFlags.IsFunctionLike | ContainerFlags.IsObjectLiteralOrClassExpressionMethod;
    }

// test
/** comment test */
// falls through
case SyntaxKind.Constructor:
case SyntaxKind.FunctionDeclaration:
case SyntaxKind.MethodSignature:
case SyntaxKind.GetAccessor:
case SyntaxKind.SetAccessor:
case SyntaxKind.CallSignature:
case SyntaxKind.JSDocSignature:
case SyntaxKind.JSDocFunctionType:
case SyntaxKind.FunctionType:
case SyntaxKind.ConstructSignature:
case SyntaxKind.IndexSignature:
case SyntaxKind.ConstructorType:
    return ContainerFlags.IsContainer | ContainerFlags.IsControlFlowContainer | ContainerFlags.HasLocals | ContainerFlags.IsFunctionLike;
```